syncapi/storage/sqlite3/memberships_table.go

```diff
@@ -63,9 +63,15 @@ const selectMembershipSQL = "" +
 	" ORDER BY stream_pos DESC" +
 	" LIMIT 1"
 
+const selectMembershipCountSQL = "" +
+	"SELECT COUNT(*) FROM (" +
+	" SELECT *, max(stream_pos) FROM syncapi_memberships WHERE room_id = $1 AND stream_pos <= $2 GROUP BY user_id" +
```

Comment:
You need an ORDER BY here like with postgres, and probably also HAVING to filter out the redundant membership rows e.g if Alice joins/leaves 5 times, use the most recent. I _think_ `HAVING(MAX(stream_pos))` should do the trick?